### ISSUE TITLE
Resize tensor in tflite

### ISSUE THREAD

**Issue by lucaswu** [tensorflow/tensorflow#16216]
hi,I save my cnn model into pb file, and  change it into tflite model.  In android I can run the model well with tensorflow lite. 

 I have a problem that the input tensor size of inference must be euqal to the input tensor size of training, because the intermediate variables  size of the model are determined by the input tensor size of training. But for my  cnn model, it can process the input tensor size that is not euqal to the input tensor size of training. 

 so  How can  i reshape the intermeidate varibales shape to match the  input tensor size of inference before inference? In other words， How can i do to let my cnn model process the input tensor whose size is not euqal to the input tensor size of traning? thx.

**Comment by andrehentz** [comment]
Call ResizeInputTensor() then call AllocateTensors().

**Comment by lucaswu** [comment]
@andrehentz , in the run() function , there  are ResizeInputTensor() and AllocateTensors() before Invoke(), so I'm confused about what your said. 
ps: I call resizeInput() before run() in my Java code.

**Comment by lucaswu** [comment]
during training, input picture size is 64x64, so the intermediate variables sizes in pb file is related to 64x64. For example, variable A's shape is tf.reshape(input,[-1,h*4,w*4,4]),for 64x64,variable A 's shape is [-1,256,256,3]. During inference, intput size is 128X128, for variable A should be [-1,512,512,3], but variable A is  still [-1,256,256,3],so the result is wrong because the first dimension is error.

**Comment by andrehentz** [comment]
Try passing [1,128,128,3] to your resizeInput call. (The special value -1 won't work in inference because the batch size needs to be fully specified)

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by drpngx** [comment]
@lucaswu did that fix your problem?

**Comment by lucaswu** [comment]
@andrehentz 's suggestion is not OK for me. After i have modify  c plus plus code of tflite, it can run now for any size. @drpngx

**Comment by drpngx** [comment]
OK, is that something that needs a fix in tflite?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrehentz** [comment]
Closing this one. Reopen if you still encounter issues.

**Comment by zjbaby** [comment]
@andrehentz 
I encount the same problem when I use mobilenet ssd models. The training input tensor size is 300x300, and I want to use different input tensor size in inference, so I use the **ResizeInputTensor()** function:
`

    std::vector<int> sizes = {1, 240, 240, 3};

    interpreter_->ResizeInputTensor(interpreter_->inputs()[0], sizes);

    if (interpreter_->AllocateTensors() != kTfLiteOk) {
        LOG(INFO) << "Failed to allocate tensors!" << "\n";
        return false;
    }`

**But I got an error in AllocateTensors():**
tensorflow/contrib/lite/kernels/reshape.cc:58 num_input_elements != num_output_elements (2700 != 4332)
Node number 36 (RESHAPE) failed to prepare.
Failed to allocate tensors!

How to fix it? if I want to change input tensor size freely, what should I do?

**Comment by zh794390558** [comment]
@andrehentz Does have same problem in using `ResizeInputTensor`

**Comment by andrehentz** [comment]
Thanks for the info. Generally speaking it should be possible to change the input tensor size just like you are doing. However, in some cases the conversion process hardcode the input shapes, which then prevents resizing at inference time. This is mostly dependent on the model architecture. In your case, I'd suggest running the conversion again, and passing --input_shapes=1,240,240,3. (Of course, this assumes you are not using a pre-converted model. If that's the case, you will need to rescale the image before passing it into the model).

**Comment by shahid14** [comment]
https://github.com/tensorflow/tensorflow/issues/22377#issuecomment-444550814

**Comment by andrehentz** [comment]
See https://github.com/tensorflow/tensorflow/issues/23600 for a possible dupe.

**Comment by shahiduos14** [comment]
Tensor can be resized in Tflite, check https://github.com/tensorflow/tensorflow/issues/23940#issuecomment-447380387

**Comment by hamlatzis** [comment]
https://github.com/tensorflow/tensorflow/issues/16216#issuecomment-361487585

@lucaswu what did you change in c++? I'm resizing the batch size, keeping width, height and channels the same, but only the input tensors change, when allocation is happening I get that a reshape node fails because the input & output shapes aren't the same.

I don't know how to find which operation doesn't modify the batch size of the output tensors

**Comment by dtlam26** [comment]
Any update on this? The result of batch are all the same at the output
